docs/admin/kubeadm.md

```diff
@@ -323,16 +323,79 @@ kind: NodeConfiguration
 caCertPath: <path|string>
 discoveryFile: <path|string>
 discoveryToken: <string>
-
-# Currently only the first server is used as a target for the cluster
-# bootstrap flow.
 discoveryTokenAPIServers:
-  - <address|string>
-  - <address|string>
-
+- <address|string>
+- <address|string>
 tlsBootstrapToken: <string>
 ```
 
+## Securing your installation even more
+
+The defaults for kubeadm may not work for everyone. This section documents how to tighten up a kubeadm install
+at the cost of some usability.
+
+### Turning off auto-approval of Node Client Certificates
+
+By default, there is an CSR auto-approver enabled that basically approves any client certificate request
+for a kubelet when a Bootstrap Token was used when authenticating. If you don't want the cluster to
+automatically approve kubelet client certs, you can turn it off by executing this command:
+
+```console
+$ kubectl delete clusterrole kubeadm:node-autoapprove-bootstrap
+```
+
+After that, `kubeadm join` will block until the admin has manually approved the CSR in flight:
+
+```console
+$ kubectl get csr
+NAME                                                   AGE       REQUESTOR                 CONDITION
+node-csr-c69HXe7aYcqkS1bKmH4faEnHAWxn6i2bHZ2mD04jZyQ   18s       system:bootstrap:878f07   Pending
+
+$ kubectl certificate approve node-csr-c69HXe7aYcqkS1bKmH4faEnHAWxn6i2bHZ2mD04jZyQ
+certificatesigningrequest "node-csr-c69HXe7aYcqkS1bKmH4faEnHAWxn6i2bHZ2mD04jZyQ" approved
+
+$ kubectl get csr
+NAME                                                   AGE       REQUESTOR                 CONDITION
+node-csr-c69HXe7aYcqkS1bKmH4faEnHAWxn6i2bHZ2mD04jZyQ   1m        system:bootstrap:878f07   Approved,Issued
+```
+
+Only after `kubectl certificate approve` has been run, `kubeadm join` can proceed.
+
+### Turning off public access to the cluster-info ConfigMap
+
+In order to achieve the joining flow using the token as the only piece of validation information, a
+public ConfigMap with some data needed for validation of the master's identity is exposed publicly by
+default. While there is no private data in this ConfigMap, some users are sensitive and wish to turn
+it off regardless. Doing so will disable the ability to use the `--discovery-token` flag of the
+`kubeadm join` flow. Here are the steps to do so:
+
+Fetch the `cluster-info` file from the API Server:
+
+```console
+$ kubectl -n kube-public get cm cluster-info -oyaml | grep "kubeconfig:" -A11 | grep "apiVersion" -A10 | sed "s/    //" | tee cluster-info.yaml
```

Comment:
How is it possible to trim four spaces in a cleaner way? This was the first I could think of